### PR TITLE
avoid unneccassary calls

### PR DESCRIPTION
**Pull request by andreasschacht**

### CODE REVIEW

File: broccoli-image-writer.js [hunk 1]
```diff
@@ -92,23 +98,19 @@ ImageResizer.prototype.generateImages = function (file, sourcePath, destinationP
...
-    let gmImage = gm(source);
     promises.push(() => {
-      return Q.all([
-        Q.ninvoke(gmImage, 'format'),
-        Q.ninvoke(gmImage, 'color'),
-        Q.ninvoke(gmImage, 'size')
-      ])
-      .then((infos) => {
-        this.insertMetadata(file, generatedFilename, width, infos);
+      imageSize.then((size) => {
```
Comment (simonihmig) [inline broccoli-image-writer.js]: Shouldn't you return the promise?

File: broccoli-image-writer.js [hunk 2]
```diff
@@ -74,9 +74,15 @@ ImageResizer.prototype.build = function () {
...
+  let gmImage = gm(path.join(sourcePath, file));
+  let imageInfos = Q.all([
+    Q.ninvoke(gmImage, 'format'),
+    Q.ninvoke(gmImage, 'color'),
+    Q.ninvoke(gmImage, 'size')
+  ]);
   this.image_options.supportedWidths.forEach((width) => {
     promises.push(() => {
-      return this.generateImage(file, sourcePath, destinationPath, width)
+      return this.generateImage(file, sourcePath, destinationPath, width, imageInfos)
```
Comment (simonihmig) [inline broccoli-image-writer.js]: I think you could as well write it like this:
```
return imageInfos.then((infos) => this.generateImage(file, sourcePath, destinationPath, width, infos))
```

So the argument to `generateImage` is the real data, not a promise. Maybe makes the `generateImage` API a bit more simple?

File: broccoli-image-writer.js (hunk 1 above)
Comment (andreasschacht) [inline broccoli-image-writer.js]: sure...

File: broccoli-image-writer.js (hunk 2 above)
Comment (andreasschacht) [inline broccoli-image-writer.js]: you're right, it's better

**Pull request merged**